govcd/metadata.go

```diff
@@ -147,3 +148,105 @@ func addMetadata(client *Client, key string, value string, requestUri string) (T
+	}
+
+	return vAppTemplate, nil
+}
+
+// AddMetadata() function calls private function addMetadata() with vAppTemplate.client and vAppTemplate.VAppTemplate.HREF
+// which adds metadata key, value pair provided as input.
+func (vAppTemplate *VAppTemplate) AddMetadataAsync(key string, value string) (Task, error) {
+	return addMetadata(vAppTemplate.client, key, value, vAppTemplate.VAppTemplate.HREF)
+}
+
+// DeleteMetadata() function calls deletes metadata depending on key provided as input from media item.
+func (vAppTemplate *VAppTemplate) DeleteMetadata(key string) error {
+	task, err := vAppTemplate.DeleteMetadataAsync(key)
+	if err != nil {
+		return err
+	}
+	err = task.WaitTaskCompletion()
+	if err != nil {
+		return fmt.Errorf("error completing delete metadata for vApp template task: %#v", err)
+	}
+
+	return nil
+}
+
+// DeleteMetadata() function calls private function deleteMetadata() with vAppTemplate.client and vAppTemplate.VAppTemplate.HREF
+// which deletes metadata depending on key provided as input from catalog item.
+func (vAppTemplate *VAppTemplate) DeleteMetadataAsync(key string) (Task, error) {
+	return deleteMetadata(vAppTemplate.client, key, vAppTemplate.VAppTemplate.HREF)
+}
+
+// GetMetadata() function calls private function getMetadata() with mediaItem.client and mediaItem.MediaItem.HREF
+// which returns a *types.Metadata struct for provided media item input.
+func (mediaItem *MediaItem) GetMetadata() (*types.Metadata, error) {
+	return getMetadata(mediaItem.client, mediaItem.MediaItem.HREF)
+}
+
+// AddMetadata() function adds metadata key, value pair provided as input.
+func (mediaItem *MediaItem) AddMetadata(key string, value string) (*MediaItem, error) {
+	task, err := mediaItem.AddMetadataAsync(key, value)
+	if err != nil {
+		return nil, err
+	}
+	err = task.WaitTaskCompletion()
+	if err != nil {
+		return nil, fmt.Errorf("error completing add metadata for media item task: %#v", err)
+	}
+
+	err = mediaItem.Refresh()
+	if err != nil {
+		return nil, fmt.Errorf("error refreshing media item: %#v", err)
+	}
+
+	return mediaItem, nil
+}
+
+// AddMetadata() function calls private function addMetadata() with mediaItem.client and mediaItem.MediaItem.HREF
+// which adds metadata key, value pair provided as input.
+func (mediaItem *MediaItem) AddMetadataAsync(key string, value string) (Task, error) {
+	return addMetadata(mediaItem.client, key, value, mediaItem.MediaItem.HREF)
+}
+
+// DeleteMetadata() function calls deletes metadata depending on key provided as input from media item.
+func (mediaItem *MediaItem) DeleteMetadata(key string) error {
+	task, err := mediaItem.DeleteMetadataAsync(key)
+	if err != nil {
+		return err
+	}
+	err = task.WaitTaskCompletion()
+	if err != nil {
+		return fmt.Errorf("error completing delete metadata for media item task: %#v", err)
+	}
+
+	return nil
+}
+
+// DeleteMetadata() function calls private function deleteMetadata() with mediaItem.client and mediaItem.MediaItem.HREF
```

Comment:
This comment should say DeleteMetadataAsync

Reply:
Changed